zerver/lib/events.py

```diff
@@ -102,13 +108,24 @@ def user_data(row: Dict[str, Any]) -> Dict[str, Any]:
             is_active = row['is_active'],
             date_joined = row['date_joined'].isoformat(),
         )
+        if for_api:
+            result['bot_type'] = row['bot_type']  # send this for humans and bots.
+        if include_delivery_emails:
+            result['delivery_email'] = row['delivery_email']
         if is_bot:
             if row['email'] in settings.CROSS_REALM_BOT_EMAILS:
                 result['is_cross_realm_bot'] = True
             elif row['bot_owner_id'] is not None:
-                result['bot_owner_id'] = row['bot_owner_id']
+                bot_owner_id = row['bot_owner_id']
+                result['bot_owner_id'] = bot_owner_id
+                if for_api:
+                    bot_owner = UserProfile.objects.get(id=bot_owner_id)
+                    result['bot_owner'] = bot_owner.email
+                    if include_delivery_emails:
+                        result['bot_owner_delivery_email'] = bot_owner.delivery_email
```

Comment:
I guess this is where you're doing that query.  I think ultimately the right fix for this is that we shouldn't be sending bot owners as an email address; we should be sending bot owner user IDs instead.  

I just checked and the mobile apps don't use the `bot_owner` field, so the right solution is to just do a prep commit series that changes `bot_owner` to be an integer ID for the bot's owner; that's a cleaner API anyway in several ways, and lets us avoid adding this mess.

Reply:
Sounds great! I was hesitant to make such a change initially because changing the API means affecting multiple people (the mobile apps as well as other (non-Zulip) developers who are using the API, like people who make bots). But I'll go ahead and make this change esp. since the bot owner's email can still be obtained from the data that this endpoint is returning without needing to make an extra request, if we at least send the id of the owner.

I'll make the change. :+1:

Reply:
Yeah, the key detail here is that none of the other apps look at the `bot_owner` field.